hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -215,253 +304,252 @@ public static void addDebugLevel(Integer level) {
+
+    if (!filename.endsWith(SST_FILE_EXTENSION)) {
+      filename += SST_FILE_EXTENSION;
+    }
+
     Options option = new Options();
     SstFileReader reader = new SstFileReader(option);
-    try {
-      reader.open(saveCompactedFilePath + filename);
-    } catch (RocksDBException e) {
-      reader.open(rocksDbPath + "/"+ filename);
+
+    File sstFile = new File(sstBackupDir + filename);
+    File sstFileInActiveDB = new File(activeDBLocationStr + filename);
+    if (sstFile.exists()) {
+      reader.open(sstBackupDir + filename);
+    } else if (sstFileInActiveDB.exists()) {
+      reader.open(activeDBLocationStr + filename);
+    } else {
+      throw new RuntimeException("Can't find SST file: " + filename);
     }
+
     TableProperties properties = reader.getTableProperties();
-    LOG.warn("getSSTFileSummary " + filename + ":: " +
-        properties.getNumEntries());
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("{} has {} keys", filename, properties.getNumEntries());
+    }
     return properties.getNumEntries();
   }
 
-  // Read the current Live manifest for a given RocksDB instance (Active or
-  // Checkpoint). Returns the list of currently active SST FileNames.
-  @SuppressFBWarnings({"NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE"})
+  /**
+   * Helper method to trim the filename retrieved from LiveFileMetaData.
+   */
+  private String trimSSTFilename(String filename) {
+    if (!filename.startsWith("/")) {
+      final String errorMsg = String.format(
+          "Invalid start of filename: '%s'. Expected '/'", filename);
+      LOG.error(errorMsg);
+      throw new RuntimeException(errorMsg);
+    }
+    if (!filename.endsWith(SST_FILE_EXTENSION)) {
+      final String errorMsg = String.format(
+          "Invalid extension of file: '%s'. Expected '%s'",
+          filename, SST_FILE_EXTENSION_LENGTH);
+      LOG.error(errorMsg);
+      throw new RuntimeException(errorMsg);
+    }
+    return filename.substring("/".length(),
+        filename.length() - SST_FILE_EXTENSION_LENGTH);
+  }
+
+  /**
+   * Read the current Live manifest for a given RocksDB instance (Active or
+   * Checkpoint).
+   * @param dbPathArg path to a RocksDB directory
+   * @return a list of SST files (without extension) in the DB.
+   */
   public HashSet<String> readRocksDBLiveFiles(String dbPathArg) {
     RocksDB rocksDB = null;
     HashSet<String> liveFiles = new HashSet<>();
-    //
-    try (final Options options =
-             new Options().setParanoidChecks(true)
-                 .setCreateIfMissing(true)
-                 .setCompressionType(CompressionType.NO_COMPRESSION)
-                 .setForceConsistencyChecks(false)) {
-      rocksDB = RocksDB.openReadOnly(options, dbPathArg);
+
+    final List<ColumnFamilyHandle> columnFamilyHandles = new ArrayList<>();
+    final List<ColumnFamilyDescriptor> cfd = new ArrayList<>();
+    cfd.add(new ColumnFamilyDescriptor(
+        "keyTable".getBytes(StandardCharsets.UTF_8)));
```

Comment:
What about the file and directory tables?

Reply:
Good catch. I will add `directoryTable` and `fileTable` as well.

I was following [PR #3885](https://github.com/apache/ozone/pull/3885/files#diff-babc9eea3d726784a8d1643c490a10c57d304232ad9c2a2cea30ebd807fe8c02R47-R49), which only tracks `keyTable` atm, meaning FSO keys and directories are ignored in that snapshot differ (for now? @nandakumar131 ).

Reply:
I believe we have separate task items for tracking support for FSO buckets. Let us Add DirectoryTable and FileTable related changes in a separate PR.